pkg/mcs/resource_manager/server/token_bukets.go

```diff
@@ -45,100 +52,229 @@ type GroupTokenBucket struct {
 	GroupTokenBucketState `json:"state,omitempty"`
 }
 
+func (gtb *GroupTokenBucket) setState(state *GroupTokenBucketState) {
+	gtb.mu.Lock()
+	defer gtb.mu.Unlock()
+
+	gtb.Tokens = state.Tokens
+	gtb.LastUpdate = state.LastUpdate
+	gtb.Initialized = state.Initialized
+	log.Info("update group token bucket state", zap.Any("state", state))
+}
+
+// TokenSlot is used to split a token bucket into multiple slots to
+// server different clients within the same resource group.
+type TokenSlot struct {
+	// settings is the token limit settings for the slot.
+	settings *rmpb.TokenLimitSettings
+	// assignTokens is the number of tokens in the slot.
+	assignTokens     float64
```

Comment:
What about `tokenCapicity` or `tokens`? `assignTokens` is a little bit confusing to me.